src/WebJobs.Extensions.DurableTask/IPlatformInformationService.cs

```diff
@@ -5,6 +5,64 @@
 
 namespace Microsoft.Azure.WebJobs.Extensions.DurableTask
 {
+    /// <summary>
+    /// Representation of the Consumption and the AppService (Dedicated or Premium) plans.
+    /// </summary>
+    public enum AppServicePlan
+    {
+        /// <summary>
+        /// Consumption App Service plan.
+        /// </summary>
+        Consumption,
+
+        /// <summary>
+        /// Non-consumption App Service plans: Dedicated, EP, etc.
+        /// </summary>
+        AppService,
+    }
+
+    /// <summary>
+    /// Representation of the supported Operating Systems.
+    /// </summary>
+    public enum OperatingSystem
+    {
+        /// <summary>
+        /// Linux OS.
+        /// </summary>
+        Linux,
+
+        /// <summary>
+        /// Windows OS
+        /// </summary>
+        Windows,
+    }
+
+    /// <summary>
+    /// Representation of supported Programming Languages.
+    /// </summary>
+    public enum ProgLanguage
```

Comment:
The name `ProgLanguage` is misleading because the programming language itself is irrelevant. What matters is the stack (.NET, Node.js, Python, PowerShell).